### ISSUE TITLE
Excel cannot open the generated file

### ISSUE THREAD

**Issue by smartcrmKandel** [ggreen86/XLXS-Workbook-Class#3]
Good morning! 
I'm trying to include the lates workbook class into our application. But the file that was created could not be red by Excel. No message from Excel, just an empty sheet. Your example is working fine. The contence of the file seems to be ok. So what goes wrong? May be I should send you the file?

Thanks for your help in advance.
Karsten

PS: My English is really poor. So please talk to me like to a 5 year old boy ;o).

**Comment by ggreen86** [comment]
Karsten--


Please send me the generated spreadsheet and the code fragment that you are using to write the data to the generated spreadsheet.  thank you.

Greg

________________________________

**Comment by smartcrmKandel** [comment]
Hello Greg,

have you got my mail with the code and the spread sheet?

Regards
Karsten

Von: ggreen86 [mailto:notifications@github.com]
Gesendet: Mittwoch, 23. August 2017 15:45
An: ggreen86/XLXS-Workbook-Class <XLXS-Workbook-Class@noreply.github.com>
Cc: Karsten Duwe <Karsten.Duwe@smartcrm.de>; Author <author@noreply.github.com>
Betreff: Re: [ggreen86/XLXS-Workbook-Class] Excel cannot open the generated file (#3)

Karsten--


Please send me the generated spreadsheet and the code fragment that you are using to write the data to the generated spreadsheet. thank you.

Greg

________________________________

**Comment by ggreen86** [comment]
Karsten--


No I did not receive any email yet.  Can you resend?  Thank you.


Greg

________________________________

**Comment by smartcrmKandel** [comment]
Greg,

here is:

       local loGridExporter

       loGridExporter = newobject([vfpxWorkbookXLSX],[vfpxWorkBookXLSX.vcx],[ivs2000.exe])
       loGridExporter.SaveGridToWorkbookEX(this,[zExcelExportNeu],.t.,[Export])

       loGridExporter = .null.

Regards
Karsten

Von: ggreen86 [mailto:notifications@github.com]
Gesendet: Mittwoch, 30. August 2017 15:32
An: ggreen86/XLXS-Workbook-Class <XLXS-Workbook-Class@noreply.github.com>
Cc: Karsten Duwe <Karsten.Duwe@smartcrm.de>; Author <author@noreply.github.com>
Betreff: Re: [ggreen86/XLXS-Workbook-Class] Excel cannot open the generated file (#3)

Karsten--


No I did not receive any email yet. Can you resend? Thank you.


Greg

________________________________

**Comment by ggreen86** [comment]
Karsten--

Thank you for sending the below code; but I will need a working example.  Can you create a test form with the grid and cursor that you want to export to a spreadsheet?

Greg

________________________________

**Comment by smartcrmKandel** [comment]
Greg,

unfortunately not. Our application creates the grids dynamically and there are some configuration
tables and a lot of code around it. Its almost impossible to run it outside our application.

Could’nt you find anything wrong in the generated spread sheet?

Karsten

Von: ggreen86 [mailto:notifications@github.com]
Gesendet: Mittwoch, 30. August 2017 19:27
An: ggreen86/XLXS-Workbook-Class <XLXS-Workbook-Class@noreply.github.com>
Cc: Karsten Duwe <Karsten.Duwe@smartcrm.de>; Author <author@noreply.github.com>
Betreff: Re: [ggreen86/XLXS-Workbook-Class] Excel cannot open the generated file (#3)

Karsten--

Thank you for sending the below code; but I will need a working example. Can you create a test form with the grid and cursor that you want to export to a spreadsheet?

Greg

________________________________

**Comment by ggreen86** [comment]
Without being able to duplicate, I cannot fix it.  There was not an attached spreadsheet that you generated.  Please attach this and then send to me.


An alternative is to create a DBF from the cursor that you create in your form and then send me the DBF.


________________________________

**Comment by smartcrmKandel** [comment]
Here it is.

Von: ggreen86 [mailto:notifications@github.com]
Gesendet: Donnerstag, 31. August 2017 16:06
An: ggreen86/XLXS-Workbook-Class <XLXS-Workbook-Class@noreply.github.com>
Cc: Karsten Duwe <Karsten.Duwe@smartcrm.de>; Author <author@noreply.github.com>
Betreff: Re: [ggreen86/XLXS-Workbook-Class] Excel cannot open the generated file (#3)

Without being able to duplicate, I cannot fix it. There was not an attached spreadsheet that you generated. Please attach this and then send to me.


An alternative is to create a DBF from the cursor that you create in your form and then send me the DBF.


________________________________

**Comment by ggreen86** [comment]
No attachment was found on the email.  Can you try to attach the spreadsheet again?


________________________________

**Comment by smartcrmKandel** [comment]
OK, one more time.

Von: ggreen86 [mailto:notifications@github.com]
Gesendet: Freitag, 1. September 2017 10:30
An: ggreen86/XLXS-Workbook-Class <XLXS-Workbook-Class@noreply.github.com>
Cc: Karsten Duwe <Karsten.Duwe@smartcrm.de>; Author <author@noreply.github.com>
Betreff: Re: [ggreen86/XLXS-Workbook-Class] Excel cannot open the generated file (#3)

No attachment was found on the email. Can you try to attach the spreadsheet again?


________________________________

**Comment by ggreen86** [comment]
I received the file via the direct send to my email account.  Thank you.  I have found and fixed the problem in the code -- had to do with the SET DECIMALS and SET SEPARATOR settings being non-USA defaults (internally everything in the XLSX is stored in USA defaults for these settings).  Since European standard is opposite to USA this was causing the column widths to be stored with the comma as the decimal.  I have now fixed this in Release 26.
